hex-grid-generator.js

```diff
@@ -1,88 +1,84 @@
-var doc = app.activeDocument
-
-// Fill color for completed hexagons
-var fillColor = new RGBColor()
-fillColor.red = 230
-fillColor.blue = 60
-fillColor.green = 60
-
-// Stroke color for completed hexagons
-var strokeColor = new RGBColor()
-strokeColor.red = 30
-strokeColor.blue = 30
-strokeColor.green = 30
-
-// Is a single thing selected?
-if (doc.selection.length !== 1) {
-  // If not, panic and yell at them
-  var text = doc.textFrames.add()
-  text.move(doc, ElementPlacement.PLACEATBEGINNING)
-  text.contents = "Draw and select a single hexagon!\nHold down the shape tooland pick the\npolygon. Click and drag while holding down the shift key.\nRotate it 90 degrees if you want it pointy at the top."
-  text.left = 40
-  text.top = -100
-  text.textRange.characterAttributes.size = 30
-} else {
-  // If yes, we'll just assume it's a hexagon.
-  var hex = doc.selection[0]
-  var hexHeight = hex.height / 2
-  var hexWidth = hex.width / 2
-  var orientation = 'horizontal'
-  if (hexHeight > hexWidth) {
-    orientation = 'vertical'
+var doc = app.activeDocument;
+
+function IsSingleItemSelected(selection) {
+  return selection.length === 1;
+}
+
+function IsAHexagonSelected(selection) {
+  return selection[0].pathPoints.length === 6;
+}
+
+function warnUser(message) {
+  var text = doc.textFrames.add();
+  text.move(doc, ElementPlacement.PLACEATBEGINNING);
+  text.contents = message;
+  text.left = 40;
+  text.top = -100;
+  text.textRange.characterAttributes.size = 30;
+}
+
+function getColStep(isVertical, hex) {
+  return isVertical ? hex.width : hex.width * 0.75;
+}
+
+function getRowStep(isVertical, hex) {
+  return isVertical ? hex.height * 0.75 : hex.height;
+}
+
+function getYOffset(isVertical, currentColumn, rowStep) {
+  return !isVertical && currentColumn % 2 == 1 && -rowStep * 0.5;
+}
+
+function getXOffset(isVertical, currentRow, colStep) {
+  return isVertical && currentRow % 2 == 1 && colStep * 0.5;
+}
+
+function getNextPosition(isVertical, rowStep, colStep, currentColumn, currentRow) {
+  var xPosition = currentColumn * colStep
+  var yPosition = currentRow * -rowStep
+  return {
+    x: xPosition + getXOffset(isVertical, currentRow, colStep),
```

Comment:
Didn't inline this because I wanted it to read "x position plus x offset" to make it simpler to visualize